tools/serverpod_cli/lib/src/internal_tools/analyze_pubspecs.dart

```diff
@@ -14,14 +69,51 @@ Future<void> performAnalyzePubspecs(bool checkLatestVersion) async {
   if (!dirPackages.existsSync() ||
       !dirTemplates.existsSync() ||
       !dirRoot.existsSync()) {
-    print('Must be run from the serverpod repository root');
-    exit(1);
+    return false;
+  }
+
+  return true;
+}
+
+void _printMissmatchedDependencies(Set<String> missmatchedDeps,
+    Map<String, List<_ServerpodDependency>> dependencies) {
+  print('Found missmatched dependencies:');
+  for (var depName in missmatchedDeps) {
+    print(depName);
+    var deps = dependencies[depName]!;
+    for (var dep in deps) {
+      print('  ${dep.version} ${dep.serverpodPackage}');
+    }
+  }
+}
+
+Set<String> _findMissmatchedDependencies(
+    Map<String, List<_ServerpodDependency>> dependencies) {
+  var missmatchedDeps = <String>{};
+  for (var depName in dependencies.keys) {
+    var deps = dependencies[depName]!;
+    String? version;
+    for (var dep in deps) {
+      if (version != null && version != dep.version) {
+        missmatchedDeps.add(depName);
+      }
+      version = dep.version;
+    }
   }
+  return missmatchedDeps;
+}
 
-  var pubspecs = _getPubspecs();
+Map<String, List<_ServerpodDependency>> _getDependencies(
+    List<File> pubspecFiles) {
   var dependencies = <String, List<_ServerpodDependency>>{};
+  for (var pubspecFile in pubspecFiles) {
+    var pubspec = tryParsePubspec(pubspecFile);
+    if (pubspec == null) {
+      print('Failed to load PUBLISHABLE_PACKAGES or the pubspec files. Are you '
```

Comment:
I'd print this error on a higher level (where you reference `PUBLISHABLE_PACKAGES`)